### PR TITLE
Added alerts and reports for fortigate firewall

### PR DESCRIPTION
**Pull request by hardikhdholariya**
- Added alerts and reports related to the FortiGate firewall data

### CODE REVIEW

File: cyences_app_for_splunk/default/savedsearches.conf [hunk 1]
```diff
@@ -1806,6 +1806,195 @@ search = | savedsearch "Palo Alto Firewall - Network Compromise - DDoS Attack Pr
...
+\
+Data Collection: Fortigate fortigate_traffic sourcetype data. (Fortinet Fortigate Add-on for Splunk)
+search = `cs_fortigate` sourcetype="fortigate_traffic" action="allowed" `cs_filter_private_ips` packets_out>0 \
+| stats count, latest(_time) as last_seen, values(dvc) as dvc, dc(dvc) as dc_dvc, dc(dest_port) as dc_dest_port, values(dstcountry) as dest_location, sum(packets_in) as packets_in, sum(packets_out) as packets_out, dc(src_ip) as dc_src_ip, dc(src_port) as dc_src_port, dc(srccountry) as dc_src_location, values(app) as app, values(rule) as firewall_rule by dest_ip \
+| `cs_fortigate_search_blocked_ip("dest_ip")` | sort -count \
+| `cs_fortigate_blocked_ip_outbound_filter`
+action.cyences_notable_event_action.products = Fortigate
+
+
+[Fortigate Firewall - Malicious IP List Gen]
```
Comment (VatsalJagani) [inline cyences_app_for_splunk/default/savedsearches.conf:1968]: Remove this.

File: cyences_app_for_splunk/default/savedsearches.conf [hunk 2]
```diff
@@ -1806,6 +1806,195 @@ search = | savedsearch "Palo Alto Firewall - Network Compromise - DDoS Attack Pr
...
+\
+Data Collection: Fortigate fortigate_traffic sourcetype data. (Fortinet Fortigate Add-on for Splunk)
+search = `cs_fortigate` sourcetype="fortigate_traffic" action="allowed" `cs_filter_private_ips` packets_in>0 \
+| stats count, latest(_time) as last_seen, values(dvc) as dvc, dc(dvc) as dc_dvc, dc(src_port) as dc_src_port, values(srccountry) as src_location, sum(packets_in) as packets_in, sum(packets_out) as packets_out, dc(dest_ip) as dc_dest_ip, values(dest_port) as dest_port, dc(dstcountry) as dc_dest_location, values(app) as app, values(rule) as firewall_rule by src_ip \
+| `cs_fortigate_search_blocked_ip("src_ip")` | sort -count \
+| `cs_fortigate_blocked_ip_inbound_filter`
+action.cyences_notable_event_action.products = Fortigate
+
+
+[Fortigate Firewall - Network Compromise - Outbound Traffic to Blocked IPs]
```
Comment (VatsalJagani) [inline cyences_app_for_splunk/default/savedsearches.conf:1952]: Can we generalize this alert with `Network Traffic` data-model?
- ensure it covers atleast Palo Alto and Fortigate logs?

File: cyences_app_for_splunk/default/savedsearches.conf [hunk 3]
```diff
@@ -1806,6 +1806,195 @@ search = | savedsearch "Palo Alto Firewall - Network Compromise - DDoS Attack Pr
...
+auto_summarize.dispatch.earliest_time = -1mon@d
+description = Fortigate Firewall prevented DDoS attack events.\
+\
+Data Collection - Fortigate Networks firewalls threat data. (Fortinet Fortigate Add-on for Splunk)
+search = `cs_fortigate` sourcetype="fortigate_utm" \
+| stats count, latest(_time) as last_seen, dc(src_port) as dc_src_port, dc(dest_ip) as dc_dest_ip, values(signature) as signature, values(dvc) as dvc, values(srccountry) as src_location, dc(dest) as dc_dest, values(action) as action by src | sort - count \
+| `cs_fortigate_ddos_prevented_filter`
+action.cyences_notable_event_action.products = Fortigate
+
+[Fortigate Firewall - Network Compromise - Inbound Traffic from Blocked IPs]
```
Comment (VatsalJagani) [inline cyences_app_for_splunk/default/savedsearches.conf:1936]: Can we generalize this alert with `Network Traffic` data-model?
- ensure it covers atleast Palo Alto and Fortigate logs?

File: cyences_app_for_splunk/default/savedsearches.conf [hunk 4]
```diff
@@ -1806,6 +1806,195 @@ search = | savedsearch "Palo Alto Firewall - Network Compromise - DDoS Attack Pr
...
+[Fortigate Firewall - Network Compromise - DDoS Attack Prevented]
+disabled = 1
+action.email.useNSSubject = 1
+alert.track = 0
+auto_summarize = 1
+auto_summarize.dispatch.earliest_time = -1mon@d
+description = Fortigate Firewall prevented DDoS attack events.\
+\
+Data Collection - Fortigate Networks firewalls threat data. (Fortinet Fortigate Add-on for Splunk)
+search = `cs_fortigate` sourcetype="fortigate_utm" \
```
Comment (VatsalJagani) [inline cyences_app_for_splunk/default/savedsearches.conf:1931]: Are we sure about the search query here?

File: cyences_app_for_splunk/default/savedsearches.conf [hunk 5]
```diff
@@ -1806,6 +1806,195 @@ search = | savedsearch "Palo Alto Firewall - Network Compromise - DDoS Attack Pr
...
+action.cyences_send_email_action = 1
+action.cyences_notable_event_action.products = Fortigate
+
+
+# Reports
+[Fortigate Firewall - Network Compromise - DDoS Attack Prevented]
+disabled = 1
+action.email.useNSSubject = 1
+alert.track = 0
+auto_summarize = 1
```
Comment (VatsalJagani) [inline cyences_app_for_splunk/default/savedsearches.conf:1926]: Why we have added this?

File: cyences_app_for_splunk/default/savedsearches.conf [hunk 6]
```diff
@@ -1806,6 +1806,195 @@ search = | savedsearch "Palo Alto Firewall - Network Compromise - DDoS Attack Pr
...
+action.cyences_notable_event_action.param.filter_macro_name = cs_fortigate_high_system_alerts_filter
+action.cyences_notable_event_action.contributing_events = `cs_fortigate` sourcetype="fortigate_event" severity IN ("high" "critical") NOT "*license*"
+action.cyences_notable_event_action.system_compromised_search = | stats sum(count) as count by dvc
+action.cyences_notable_event_action.system_compromised_drilldown = `cs_fortigate` sourcetype="fortigate_event" severity IN ("high" "critical") NOT "*license*" dvc=$row.dvc$
+action.cyences_send_email_action = 1
+action.cyences_notable_event_action.products = Fortigate
+
+
+# Reports
+[Fortigate Firewall - Network Compromise - DDoS Attack Prevented]
```
Comment (VatsalJagani) [inline cyences_app_for_splunk/default/savedsearches.conf:1922]: Kindly copy another alert in savedsearches.conf to ensure we are not missing any important parameter and not have any extra parameter.

File: cyences_app_for_splunk/default/savedsearches.conf [hunk 7]
```diff
@@ -1806,6 +1806,195 @@ search = | savedsearch "Palo Alto Firewall - Network Compromise - DDoS Attack Pr
...
+action.cyences_notable_event_action = 1
+action.cyences_notable_event_action.param.filter_macro_name = cs_fortigate_high_system_alerts_filter
+action.cyences_notable_event_action.contributing_events = `cs_fortigate` sourcetype="fortigate_event" severity IN ("high" "critical") NOT "*license*"
+action.cyences_notable_event_action.system_compromised_search = | stats sum(count) as count by dvc
+action.cyences_notable_event_action.system_compromised_drilldown = `cs_fortigate` sourcetype="fortigate_event" severity IN ("high" "critical") NOT "*license*" dvc=$row.dvc$
+action.cyences_send_email_action = 1
+action.cyences_notable_event_action.products = Fortigate
+
+
+# Reports
```
Comment (VatsalJagani) [inline cyences_app_for_splunk/default/savedsearches.conf:1921]: Is this report or alert?

File: cyences_app_for_splunk/default/savedsearches.conf [hunk 8]
```diff
@@ -1806,6 +1806,195 @@ search = | savedsearch "Palo Alto Firewall - Network Compromise - DDoS Attack Pr
...
+| table _time, subtype, signature, cyences_severity, action, description, dvc \
+| `cs_human_readable_time_format(_time, event_time)` \
+| `cs_fortigate_high_system_alerts_filter`
+action.cyences_notable_event_action = 1
+action.cyences_notable_event_action.param.filter_macro_name = cs_fortigate_high_system_alerts_filter
+action.cyences_notable_event_action.contributing_events = `cs_fortigate` sourcetype="fortigate_event" severity IN ("high" "critical") NOT "*license*"
+action.cyences_notable_event_action.system_compromised_search = | stats sum(count) as count by dvc
+action.cyences_notable_event_action.system_compromised_drilldown = `cs_fortigate` sourcetype="fortigate_event" severity IN ("high" "critical") NOT "*license*" dvc=$row.dvc$
+action.cyences_send_email_action = 1
+action.cyences_notable_event_action.products = Fortigate
```
Comment (VatsalJagani) [inline cyences_app_for_splunk/default/savedsearches.conf:1918]: These all alerts should be visible on the Overview page, is it?

File: cyences_app_for_splunk/default/savedsearches.conf [hunk 9]
```diff
@@ -1806,6 +1806,195 @@ search = | savedsearch "Palo Alto Firewall - Network Compromise - DDoS Attack Pr
...
+\
+Data Collection - Fortigate firewalls system/events data (Fortinet Fortigate Add-on for Splunk).
+dispatch.earliest_time = -32m@m
+dispatch.latest_time = -2m@m
+display.general.type = statistics
+display.page.search.tab = statistics
+display.page.search.mode = fast
+request.ui_dispatch_app = cyences_app_for_splunk
+request.ui_dispatch_view = search
+search = `cs_fortigate` sourcetype="fortigate_event" severity IN ("critical", "high") NOT "*license*" \
```
Comment (VatsalJagani) [inline cyences_app_for_splunk/default/savedsearches.conf:1906]: Are we sure about this search?

File: cyences_app_for_splunk/default/savedsearches.conf [hunk 10]
```diff
@@ -1806,6 +1806,195 @@ search = | savedsearch "Palo Alto Firewall - Network Compromise - DDoS Attack Pr
...
+\
+Data Collection - Fortigate Networks firewalls traffic data (Fortinet Fortigate Add-on for Splunk).
+dispatch.earliest_time = -32m@m
+dispatch.latest_time = -2m@m
+display.general.type = statistics
+display.page.search.tab = statistics
+display.page.search.mode = fast
+request.ui_dispatch_app = cyences_app_for_splunk
+request.ui_dispatch_view = search
+search = `cs_fortigate` sourcetype="fortigate_utm" severity IN ("high" "critical") \
```
Comment (VatsalJagani) [inline cyences_app_for_splunk/default/savedsearches.conf:1867]: Are we sure about this search?

File: cyences_app_for_splunk/default/savedsearches.conf [hunk 11]
```diff
@@ -1806,6 +1806,195 @@ search = | savedsearch "Palo Alto Firewall - Network Compromise - DDoS Attack Pr
...
+\
+Data Collection - Fortigate Networks firewalls traffic data (Fortinet Fortigate Add-on for Splunk).
+dispatch.earliest_time = -62m@m
+dispatch.latest_time = -2m@m
+display.general.type = statistics
+display.page.search.tab = statistics
+display.page.search.mode = fast
+request.ui_dispatch_app = cyences_app_for_splunk
+request.ui_dispatch_view = search
+search = `cs_fortigate` (sourcetype="fortigate_traffic" OR sourcetype="fortigate_utm") dest_ip="72.5.65.111" \
```
Comment (VatsalJagani) [inline cyences_app_for_splunk/default/savedsearches.conf:1833]: Are we sure about this IP for fortigate for DNS sinkhole?

**Review by VatsalJagani — CHANGES_REQUESTED** [review]

File: cyences_app_for_splunk/default/savedsearches.conf (hunk 11 above)
Comment (hardikhdholariya) [reply cyences_app_for_splunk/default/savedsearches.conf:1833]: No, this needs to decide. I kept just for testing.

File: cyences_app_for_splunk/default/savedsearches.conf (hunk 10 above)
Comment (hardikhdholariya) [reply cyences_app_for_splunk/default/savedsearches.conf:1867]: I googled for fortigate_utm logs and it is related to threat. UTM (unified threat management) hence similar to palo, this alert generates for high and critical severity threat events.

File: cyences_app_for_splunk/default/savedsearches.conf (hunk 9 above)
Comment (hardikhdholariya) [reply cyences_app_for_splunk/default/savedsearches.conf:1906]: "fortigate_event" sourcetype is for system events (found from its data model which is categorize in datasets)

Not sure about license info. currently, no any event available related to license.

File: cyences_app_for_splunk/default/savedsearches.conf (hunk 10 above)
Comment (VatsalJagani) [reply cyences_app_for_splunk/default/savedsearches.conf:1867]: cool.

**Review by VatsalJagani — APPROVED** [review]

**Pull request merged**